kubernetes/conanfile.py

```diff
@@ -0,0 +1,66 @@
+
+    # Optional metadata
+    license = "Apache-2.0"
+    url = "https://github.com/kubernetes-client/c"
+    description = "Official C client library for Kubernetes"
+    topics = ("kubernetes", "k8s", "kubernetes-client", "k8s-client")
+
+    # Binary configuration
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False], "openssl_shared":[True, False], "openssl_version": ["1", "3"], "curl_version": ["7", "8"]}
+    default_options = {"shared": False, "fPIC": True, "openssl_shared": True, "openssl_version": "3", "curl_version": "8"}
+
+    # Sources are located in the same place as this recipe, copy them to the recipe
+    exports_sources = "config.h.in", "ConfigureChecks.cmake", "PreTarget.cmake", "PostTarget.cmake", "CMakeLists.txt", "src/*", "external/*", "api/*", "model/*", "include/*", "config/*", "watch/*", "websocket/*"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            self.options.rm_safe("fPIC")
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        self.options["openssl/*"].shared = self.options.openssl_shared
+
+    def layout(self):
+        cmake_layout(self)
+
+    def generate(self):
+        deps = CMakeDeps(self)
+        deps.generate()
+        tc = CMakeToolchain(self)
+        tc.generate()
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        copy(self, "*.h", src=join(self.source_folder, "api"), dst=join(self.package_folder, "include/kubernetes/api"), keep_path=False)
+        copy(self, "*.h", src=join(self.source_folder, "model"), dst=join(self.package_folder, "include/kubernetes/model"), keep_path=False)
+        copy(self, "*.h", src=join(self.source_folder, "config"), dst=join(self.package_folder, "include/kubernetes/config"), keep_path=False)
+        copy(self, "*.h", src=join(self.source_folder, "include"), dst=join(self.package_folder, "include/kubernetes/include"), keep_path=False)
+        copy(self, "*.h", src=join(self.source_folder, "websocket"), dst=join(self.package_folder, "include/kubernetes/websocket"), keep_path=False)
+        copy(self, "*.h", src=join(self.source_folder, "external"), dst=join(self.package_folder, "include/kubernetes/external"), keep_path=False)
+        copy(self, "*.h", src=join(self.source_folder, "watch"), dst=join(self.package_folder, "include/kubernetes/watch"), keep_path=False)
+        cmake = CMake(self)
+        cmake.install()
+
+    def package_info(self):
+        self.cpp_info.libs = ["kubernetes"]
+
+    def requirements(self):
+        self.requires("libcurl/[~{}]".format(self.options.curl_version), transitive_headers=True)
+        self.requires("openssl/[~{}]".format(self.options.openssl_version), force=True)
+        self.requires("libwebsockets/[^4]", transitive_headers=True)
+        self.requires("libyaml/[^0.2]")
```

Comment:
Why did the minimum version change from v0.2.5 to v0.2 for libyaml ?

Reply:
my reasoning was because of conan's version range expressions.
https://docs.conan.io/2/tutorial/versioning/version_ranges.html#range-expressions

`self.requires("libwebsockets/[^4.2]")` means only the highest version in 4.x is supported, 5.x (in future) will not be supported.
`self.requires("libyaml/[^0.2.5]")` means only the highest version in 0.2.x is supported, 0.3.x (in future) will not be supported.

if that's okay, then I can update it like this.

Reply:
I prefer  `^4.2` and `^0.2.5` because we can lock the current major version and only bump the minor version.
If in the future we need to support new major versions, we can update the document then. Thank you for your update.